adapter/internal/discovery/xds/server.go

```diff
@@ -171,60 +167,91 @@ func GetEnforcerApplicationKeyMappingCache() wso2_cache.SnapshotCache {
 }
 
 // UpdateAPI updates the Xds Cache when OpenAPI Json content is provided
-func UpdateAPI(byteArr []byte, upstreamCerts []byte, apiType string, environments []string) {
-	var apiMapKey string
+func UpdateAPI(apiContent config.APIContent) {
 	var newLabels []string
 	var mgwSwagger mgw.MgwSwagger
-	vhost := "default" //TODO: (SuKSW) update once vhost feature added
+	if len(apiContent.Environments) == 0 {
+		apiContent.Environments = append(apiContent.Environments, "default")
+	}
 
 	//TODO: (VirajSalaka) Optimize locking
 	var l sync.Mutex
 	l.Lock()
 	defer l.Unlock()
 
-	if apiType == mgw.HTTP {
-		mgwSwagger = operator.GetMgwSwagger(byteArr)
-	} else if apiType == mgw.WS {
-		mgwSwagger = operator.GetMgwSwaggerWebSocket(byteArr)
+	if apiContent.APIType == mgw.HTTP {
+		mgwSwagger = operator.GetMgwSwagger(apiContent.APIDefinition)
+		mgwSwagger.SetName(apiContent.Name)
+		mgwSwagger.SetVersion(apiContent.Version)
+	} else if apiContent.APIType == mgw.WS {
+		mgwSwagger = operator.GetMgwSwaggerWebSocket(apiContent.APIDefinition)
 	} else {
 		// Unreachable else condition. Added in case previous apiType check fails due to any modifications.
 		logger.LoggerXds.Error("API type not currently supported with WSO2 Microgateway")
 	}
-	apiMapKey = vhost + ":" + mgwSwagger.GetTitle() + ":" + mgwSwagger.GetVersion()
-	existingMgwSwagger, exists := apiMgwSwaggerMap[apiMapKey]
-	if exists {
-		if reflect.DeepEqual(mgwSwagger, existingMgwSwagger) {
-			logger.LoggerXds.Infof("API %v already exists. No changes to apply.", apiMapKey)
-			return
-		}
-	}
-	apiMgwSwaggerMap[apiMapKey] = mgwSwagger
+	apiIdentifier := apiContent.VHost + ":" + apiContent.Name + ":" + apiContent.Version // TODO: (SuKSW) update once vhost feature added
```

Comment:
shouldn't we use `GenerateIdentifierForAPI` here

Reply:
Adding this locally, to include in the next PR of the same feature